### PR TITLE
Configure ts-jest so we get type-checking in spec runs

### PR DESCRIPTION
**Pull request by pvenable**

### CODE REVIEW

File: dist/AccordionSection/AccordionHeader.d.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ declare function AccordionHeader({ isOpen, note, onClick, panelId, text, iconRig
     panelId: any;
     text: any;
     iconRight: any;
-}): JSX.Element;
+}): import("@emotion/react/jsx-runtime").JSX.Element;
```
Comment (pvenable) [inline dist/AccordionSection/AccordionHeader.d.ts:10]: This is due to `"jsxImportSource": "@emotion/react"` in `tsconfig.json`, but ultimately this is effectively an alias for the same type in React.

File: tsconfig.json [hunk 2]
```diff
@@ -6,9 +6,12 @@
     "declarationDir": "dist",
     "esModuleInterop": true,
     "jsx": "react-jsx",
+    "jsxImportSource": "@emotion/react",
```
Comment (pvenable) [inline tsconfig.json:9]: This makes the `css` prop typecheck.

File: tsconfig.json [hunk 3]
```diff
@@ -6,9 +6,12 @@
...
+    "lib": ["dom", "es2017"],
     "noEmit": true,
     "skipLibCheck": true,
-    "strict": true
+    "strict": true,
+    "types": ["jest", "jest-enzyme", "jest-plugin-context"]
```
Comment (pvenable) [inline tsconfig.json:14]: Oddly, I could not get some of `jest-enzyme`'s matchers to type-check consistently without adding `jest-enzyme` to `types` here.  And now that we're specifying `types` explicitly, we had to add `jest` and `jest-plugin-context` here as well, along with `"lib": ["dom", "es2017"]` above.

File: jest.config.js [hunk 4]
```diff
@@ -1,7 +1,8 @@
 module.exports = {
+  preset: 'ts-jest/presets/js-with-babel',
   testEnvironment: 'jsdom',
   verbose: true,
-  snapshotSerializers: ['@emotion/snapshot-serializer'],
+  snapshotSerializers: ['@emotion/jest/enzyme-serializer'],
```
Comment (pvenable) [inline jest.config.js:5]: I noticed emotion recommends this serializer when using enzyme, so I switched it while I was at it.

File: package.json [hunk 5]
```diff
@@ -10,7 +10,7 @@
     "build-storybook": "build-storybook",
     "lint": "yarn eslint && stylelint src",
     "eslint": "eslint src stories .storybook",
-    "test:ci": "yarn typecheck && yarn test && yarn lint",
+    "test:ci": "yarn test && yarn lint",
```
Comment (pvenable) [inline package.json:13]: We don't need to `yarn typecheck` separately here now as testing now does type-checking.

File: src/Avatar/__tests__/StyledAvatar.test.js [hunk 6]
```diff
@@ -19,7 +20,7 @@ describe('<StyledAvatar />', () => {
           />
         );
 
-        expect(avatar.find('Flex')).toHaveProp({
+        expect(avatar.find(Flex)).toHaveProp({
```
Comment (pvenable) [inline src/Avatar/__tests__/StyledAvatar.test.js:23]: A few cases of `find` by string stopped working, and needed to be switched to `find` by ComponentClass.

**Review by xxx — APPROVED** [review]

**Pull request merged**